packages/kubernetes-api/src/paging-metadata-impl.ts

```diff
@@ -0,0 +1,158 @@
+    if (this.numPages() <= this._current) return
+
+    const pagingRef = this.pagingRefs[this._current]
+    pagingRef.continue = continueRef
+  }
+
+  refresh() {
+    if (this.numPages() <= this._current) return
+
+    const pagingRef = this.pagingRefs[this._current]
+    pagingRef.count = -1
+
+    // Retain the continueRef since that was set by previous to current
+
+    if (this._current + 1 < this.numPages()) {
+      // Remove all next paging refs as they are potentially out-of-date
+      this.pagingRefs = this.pagingRefs.slice(0, this._current + 1)
+    }
+  }
+
+  resolveContinue(count: number, required: number, continueRef: string | undefined): string | undefined {
+    if (!this.isContinueRef(continueRef)) return undefined
+
+    const enoughPods = required - count === 0
+    if (!enoughPods) {
+      // Not enough pods were found to satisfy the required number
+      // so need to return the continueRef to find some more
+      return continueRef
+    }
+
+    // The correct number of pods has been found.
+    // Add the continue reference to the 'next' page as there are possibly more
+    // jolokia pods to be requested when the next() page is required
+
+    const next = this._current + 1
+    if (this.numPages() > next) {
+      // Replace an existing next page with a new continue ref
+
+      // Remove all paging refs after current since a new query has been
+      // made and a new continue ref is being added
+      this.pagingRefs = this.pagingRefs.slice(0, next)
+    }
+
+    // Adds a new 'next' paging reference
+    this.pagingRefs.push({ count: -1, continue: continueRef })
+
+    // Enough pods found, don't return the continueRef as saved for next page
+    return undefined
+  }
+
+  addPage(count: number, required: number, continueRef: string | undefined): string | undefined {
+    const pagingRefLen = this.pagingRefs.push({ count: count })
+    this._current = pagingRefLen - 1
+
+    return this.resolveContinue(count, required, continueRef)
+  }
+
+  count(index?: number): number {
+    if (!index) index = this._current
+
+    if (this.numPages() <= index) return -1
+
+    const pagingRef = this.pagingRefs[index]
+    return pagingRef.count
+  }
+
+  setCount(count: number) {
+    if (this.numPages() <= this._current) return
+
+    const pagingRef = this.pagingRefs[this._current]
+    pagingRef.count = count
+  }
+
+  continueRef(index?: number): string | undefined {
+    if (!index) index = this._current
+
+    if (this.numPages() <= index) return undefined
+
+    const pagingRef = this.pagingRefs[index]
+    return pagingRef.continue
+  }
+
+  hasPrevious(): boolean {
+    if (this._current === 0) return false
+
+    const pagingRef = this.pagingRefs[this._current - 1]
+    return isObject(pagingRef)
+  }
+
+  /**
+   * Moves current back 1 to get the previous paging metadata
+   */
+  previous() {
+    if (this._current === 0) return
+
+    if (!this.hasPrevious()) return
+
+    --this._current
```

Comment:
what's the difference between `--this._current` vs `this._current--`?  if there's no difference, then I'd prefer the latter.

Reply:
```
If used postfix, with operator after operand (for example, x--), the decrement operator decrements and returns the value before decrementing.

If used prefix, with operator before operand (for example, --x), the decrement operator decrements and returns the value after decrementing.
```

In this context, since there is no variable then its make no difference. Prefixing has just been my default. Will update.

Reply:
As it happens, that is a class that is no longer used (was conceived with a different solution to the paging problem which I ultimately ditched).